src/logmessage.cpp

```diff
@@ -136,14 +140,15 @@ namespace g3 {
 
    LogMessage::LogMessage(const std::string& file, const int line,
                           const std::string& function, const LEVELS& level)
-      : _timestamp(g3::systemtime_now())
-      , _call_thread_id(std::this_thread::get_id())
+      : _call_thread_id(std::this_thread::get_id())
       , _microseconds(microsecondsCounter())
       , _file(splitFileName(file))
       , _line(line)
       , _function(function)
       , _level(level)
-   {}
+   {
+      clock_gettime(CLOCK_REALTIME, &_timestamp);
```

Comment:
Since c++11 I think you can use timespec_get(...) on both Linux and Windows:

http://en.cppreference.com/w/c/chrono/timespec_get